src/components/content/immunizations/patient-immunizations.tsx

```diff
@@ -23,13 +24,19 @@ function PatientImmunizationsComponent({ className }: PatientImmunizationsProps)
   const containerRef = useRef<HTMLDivElement>(null);
   const breakpoints = useBreakpoints(containerRef);
   const { featureFlags } = useCTW();
-  const patientImmunizationsQuery = usePatientImmunizations();
+  const query = usePatientImmunizations();
   const openDetails = useResourceDetailsDrawer({
     header: (m) => m.description,
     details: immunizationData,
     getSourceDocument: true,
   });
 
+  useEffect(() => {
+    if (!query.isLoading) {
+      Telemetry.reportZAPRecordCount("immunization", query.data?.length);
+    }
+  }, [query.isLoading, query.data]);
```

Comment:
same here (checking query.data exists before calling reportZAPRecordCount

```suggestion
  useEffect(() => {
    if (!query.isLoading && query.data) {
      Telemetry.reportZAPRecordCount("immunization", query.data.length);
    }
  }, [query.isLoading, query.data]);
```